velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -679,6 +681,85 @@ TpchPlan TpchQueryBuilder::getQ14Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ16Plan() const {
+  std::vector<std::string> partColumns = {
+      "p_brand", "p_type", "p_size", "p_partkey"};
+  std::vector<std::string> supplierColumns = {"s_suppkey", "s_comment"};
+  std::vector<std::string> partsuppColumns = {"ps_partkey", "ps_suppkey"};
+
+  const auto partSelectedRowType = getRowType(kPart, partColumns);
+  const auto& partFileColumns = getFileColumnNames(kPart);
+  const auto supplierSelectedRowType = getRowType(kSupplier, supplierColumns);
+  const auto& supplierFileColumns = getFileColumnNames(kSupplier);
+  const auto partsuppSelectedRowType = getRowType(kPartsupp, partsuppColumns);
+  const auto& partsuppFileColumns = getFileColumnNames(kPartsupp);
+
+  const std::string partBrandFilter = "p_brand <> 'Brand#45'";
+  const std::string partTypeFilter = "p_type NOT LIKE 'MEDIUM POLISHED%'";
+  const std::string partSizeFilter = "p_size in (49, 14, 23, 45, 19, 3, 36, 9)";
+  const std::string commentFilter = "s_comment LIKE '%Customer%Complaints%'";
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId partScanNodeId;
+  core::PlanNodeId supplierScanNodeId;
+  core::PlanNodeId partsuppScanNodeId;
+
+  auto part = PlanBuilder(planNodeIdGenerator, pool_.get())
+                  .tableScan(
+                      kPart,
+                      partSelectedRowType,
+                      partFileColumns,
+                      {partSizeFilter},
+                      partTypeFilter)
+                  .capturePlanNodeId(partScanNodeId)
+                  .filter(partBrandFilter)
+                  .planNode();
+
+  auto supplier = PlanBuilder(planNodeIdGenerator)
+                      .tableScan(
+                          kSupplier,
+                          supplierSelectedRowType,
+                          supplierFileColumns,
+                          {},
+                          commentFilter)
```

Comment:
commentFilter should also be part of the subfieldFilters

Reply:
Since commentFilter uses 'like', it should be in remaining filter